### DISCUSSION TITLE
Modbus RS485 to ETH (B)

### DISCUSSION THREAD

**Discussion by ellingramstad** [wills106/homeassistant-solax-modbus#695] [Q&A]
Hello, I have purchased some RS485 to ETH(B), but I am having trouble establishing communication with it. I am using Vircom to set it up, but when I try to establish contact with it using Home Assistant, I do get in touch with it. It lights up blue for 2 seconds and then returns to yellow on the LINK LED - the ACT light faintly turns red. I have not connected any RS485 signal yet, as I want to set up the link/communication with the device first. Is it necessary to connect an RS485 signal for the device to light up blue on the LINK LED?

**Comment by wills106** [comment]
I don't connect to my Inverter over RS485. But I would assume you need it connected to the Inverter to establish a connection.

R485 - Ethernet devices are invisible to the Integration. They just translate between the different types of Modbus.

**Comment by ellingramstad** [comment]
Thank you very much for your response!
Okay, thanks. I've got the Waveshare link to work, but when I run the Solax Modbus program, I don't see any devices or readings. I've entered the correct bandwidth, etc., and everything looks right. Your Home Assistant program communicates when I start it, but no data is coming through. What could be wrong? I'm using the Growatt MID 6KTL3-XL.
